### ISSUE TITLE
"Fall back to string conversion error" warning when <NA> (NACharacterType) present in pandas column

### ISSUE THREAD

**Issue by krassowski** [rpy2/rpy2#751]
**Describe the issue or bug**

Conversion of a column with R's NA (NACharacterType) from pandas to R generates a warning ("error",  fallbacks to string conversion), but actually it seems to work fine. The error message may not be needed.

**To Reproduce**

Using IPython:

```python
import rpy2
%load_ext rpy2.ipython
```

```ipython
%%R -o df
df = data.frame(x=c(1,2,3), y=c('a', 'b', NA))
df
```

```
  x    y
1 1    a
2 2    b
3 3 <NA>
```

```python
type(df.loc['3', 'y'])
```

`rpy2.rinterface_lib.sexp.NACharacterType
`
```
df2 = df
%R -i df2
```

Error is shown, but everything seems to work perfectly otherwise. If `y` column is numeric to begin with, no error is shown.

**Expected behavior**

The round trip from R to Python and back to R is smooth and without errors.

**Error**

```
pandas2ri.py: Error while trying to convert the column "y". Fall back to string conversion. The error is: Series can only be of one type, or None (and here we have <class 'str'> and <class 'rpy2.rinterface_lib.sexp.NACharacterType'>).
```

**Additional context**

Reproduced in:

- rpy2: 3.3.2 and 3.3.6
- pandas: 1.1.4
- R: 4.0.2
- Python: 3.7.6 and 3.8.1

**Comment by lgautier** [comment]
It could be something around the conversion of factors. Is the issue also present with?

```
%%R -o df
df = data.frame(x=c(1,2,3), y=c('a', 'b', NA), stringAsFactors=FALSE)
df
```

**Comment by lgautier** [comment]
Scratch that. This is not the issue. The error message says:
```
(and here we have <class 'str'> and <class 'rpy2.rinterface_lib.sexp.NACharacterType'>)
```

The check for homogeneity of type within a column trips on Python strings being of different type than `NACharacterType`. That seems like a bug.

**Comment by lgautier** [comment]
I think that the issue is when converting from R to pandas in the first place. The NA value in R should be become `None` (there is an `NA` in `pandas` but it is marked "experimental", so may be not a good idea for now).

**Comment by lgautier** [comment]
Fixed with 32e284d341525be8b5059c26ee2a5a578b7d22af.
